crates/bevy_ecs/src/component.rs

```diff
@@ -586,61 +586,72 @@ impl Components {
 #[derive(Copy, Clone, Debug)]
 pub struct Tick {
-    pub(crate) tick: u32,
+    tick: u32,
 }
 
 impl Tick {
+    /// The maximum relative age for a change tick.
+    /// The value of this is equal to [`crate::change_detection::MAX_CHANGE_AGE`].
+    ///
+    /// Since change detection will not work for any ticks older than this,
+    /// ticks are periodically scanned to ensure their relative values are below this.
+    pub const MAX: Self = Self::new(MAX_CHANGE_AGE);
+
     pub const fn new(tick: u32) -> Self {
         Self { tick }
     }
 
+    /// Gets the value of this change tick.
+    #[inline]
+    pub const fn get(self) -> u32 {
+        self.tick
+    }
+
+    /// Sets the value of this change tick.
+    #[inline]
+    pub fn set(&mut self, tick: u32) {
+        self.tick = tick;
+    }
+
     #[inline]
-    /// Returns `true` if this `Tick` occurred since the system's `last_change_tick`.
+    /// Returns `true` if this `Tick` occurred since the system's `last_run`.
     ///
-    /// `change_tick` is the current tick of the system, used as a reference to help deal with wraparound.
-    pub fn is_newer_than(&self, last_change_tick: u32, change_tick: u32) -> bool {
-        // This works even with wraparound because the world tick (`change_tick`) is always "newer" than
-        // `last_change_tick` and `self.tick`, and we scan periodically to clamp `ComponentTicks` values
+    /// `this_run` is the current tick of the system, used as a reference to help deal with wraparound.
+    pub fn is_newer_than(self, last_run: Tick, this_run: Tick) -> bool {
+        // This works even with wraparound because the world tick (`this_run`) is always "newer" than
+        // `last_run` and `self.tick`, and we scan periodically to clamp `ComponentTicks` values
         // so they never get older than `u32::MAX` (the difference would overflow).
         //
         // The clamp here ensures determinism (since scans could differ between app runs).
-        let ticks_since_insert = change_tick.wrapping_sub(self.tick).min(MAX_CHANGE_AGE);
-        let ticks_since_system = change_tick
-            .wrapping_sub(last_change_tick)
-            .min(MAX_CHANGE_AGE);
+        let ticks_since_insert = this_run.relative_to(self).tick.min(MAX_CHANGE_AGE);
+        let ticks_since_system = this_run.relative_to(last_run).tick.min(MAX_CHANGE_AGE);
 
         ticks_since_system > ticks_since_insert
     }
 
-    pub(crate) fn check_tick(&mut self, change_tick: u32) {
-        let age = change_tick.wrapping_sub(self.tick);
-        // This comparison assumes that `age` has not overflowed `u32::MAX` before, which will be true
-        // so long as this check always runs before that can happen.
-        if age > MAX_CHANGE_AGE {
-            self.tick = change_tick.wrapping_sub(MAX_CHANGE_AGE);
-        }
+    /// Returns a change tick representing the relationship between `self` and `other`.
+    pub fn relative_to(self, other: Self) -> Self {
```

Comment:
This doesn't feel like it should return a Tick to me. A tick is an instant, but this is a Duration.

Reply:
I also don't like that it returns wrapped values instead of negative ones. Fine as an implementation detail, but less so as a public api

Reply:
Agreed on these comments, I would actually like to see this return a https://doc.rust-lang.org/std/cmp/enum.Ordering.html

Reply:
I'll just make it private. This method won't be useful if it returns anything other than a tick.